### PR TITLE
Add Authenticator for Symfony 5.1+

### PR DESCRIPTION
**Pull request by fabpot**

### CODE REVIEW

File: src/DependencyInjection/Security/Factory/ConnectFactory.php [hunk 1]
```diff
@@ -20,7 +21,7 @@
 /**
  * @author Marc Weistroff <marc.weistroff@sensiolabs.com>
  */
-class ConnectFactory extends AbstractFactory
+class ConnectFactory extends AbstractFactory implements AuthenticatorFactoryInterface
```
Comment (wouterj) [inline src/DependencyInjection/Security/Factory/ConnectFactory.php:24]: As the authenticator also provides an entry point, you must implement `EntryPointFactoryInterface` in this factory.

If this bundle still allows <5.1 (I think it does, as `symfonycorp/connect` allows it), you should create a new class that extends this one and implements both new interfaces. See https://github.com/scheb/2fa/blob/master/src/bundle/SchebTwoFactorBundle.php#L39-L44 for an example.

File: src/Resources/config/security.xml [hunk 2]
```diff
@@ -40,5 +40,17 @@
         </service>
 
         <service id="SymfonyCorp\Bundle\ConnectBundle\Controller\OAuthController" alias="symfony_connect.oauth_controller" public="true" />
+
+       <service id="connect.authenticator" class="SymfonyCorp\Connect\Security\ConnectAuthenticator">
```
Comment (wouterj) [inline src/Resources/config/security.xml:44]:
```suggestion
        <service id="connect.authenticator" class="SymfonyCorp\Connect\Security\ConnectAuthenticator">
```

(sorry, couldn't resist)

Comment (fabpot) [reply src/Resources/config/security.xml:44]: Well done sir :)

File: src/DependencyInjection/Security/Factory/ConnectFactory.php (hunk 1 above)
Comment (fabpot) [reply src/DependencyInjection/Security/Factory/ConnectFactory.php]: Both should be fixed now.

**Pull request merged**